src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs

```diff
@@ -27,33 +31,51 @@ internal abstract partial class AbstractNavigateToSearchService : INavigateToSea
         public async Task<ImmutableArray<INavigateToSearchResult>> SearchDocumentAsync(
             Document document, string searchPattern, IImmutableSet<string> kinds, CancellationToken cancellationToken)
         {
-            var client = await TryGetRemoteHostClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
-            if (client == null)
+            var client = await RemoteHostClient.TryGetClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
+            if (client != null)
             {
-                return await SearchDocumentInCurrentProcessAsync(
```

Comment:
why these cahnges?  the code was really clear before to me.  we do a split between making a local or remote call.  Now, this method gets a lot more complex and it's less clear to see what's going on.

Reply:
If you waited on the PR to be ready for review you could have read this in the description of the PR that I'll fill in once the PR is read for review. In any case, this change unifies the calls to remote services across all features. Currently we have at least a few patterns used across different services and some of them are also incorrectly handling the case when OOP isn't available (is disabled). When that happens we should consistently fall back to in-proc implementation.

Reply:
I didn't know it wasn't ready.  There is nothing to indicate that on my end:

![image](https://user-images.githubusercontent.com/4564579/70273491-0c111700-175f-11ea-9a30-3d0a754014a9.png)

> Currently we have at least a few patterns used across different services and some of them are also incorrectly handling the case when OOP isn't available (is disabled). When that happens we should consistently fall back to in-proc implementation.

Note that this should be handled with that existing pattern.  i.e. we call:

```c#
var client = await RemoteHostClient.TryGetClientAsync
```

So if OOP isn't available, you shouldn't get a client back, so it would naturally call the in-proc version.

Reply:
![image](https://user-images.githubusercontent.com/41759/70273697-7c1f9d00-175f-11ea-8ea7-1de5c5aa2a77.png)

Reply:
> Note that this should be handled with that existing pattern. i.e. we call:

It is not completely. There are cases where the remote call returns null because it can't establish a connection to OOP. That case wasn't handled properly in a few call sites. Hence the change.

Reply:
I'd prefer those cases be fixed up rather than making all these guys no longer have the clean split of "Call out to local/remote method".  I'm fine with unifying them.  However, the pattern for the features i wrote was very clear and very intentional here.  You get the connectoin to the remote side.  If you can, you call into it.  If you can't, you call into the "perform locally" method.

Reply:
It's still a clean split. The pattern is:
```
FeatureEntryPoint()
{
  if (client != null) { var result = client.RemoteCall(); if (result.HasValue) return result.Value; } 
  inProcImpl(); 
}
```

I'm just making the current design correct and consistent. The current design is that the remote call might fail and not return result, so the services must handle that. 

If we remove the possibility of the remote call failing then we can follow up and remove the fallback like so:

```
FeatureEntryPoint()
{
  if (client != null) { return client.RemoteCall(); } 
  inProcImpl(); 
}
```

I don't see much value in extracting that remote call to a separate method but we can do that as well, if that's preferred.